File: MAP(Advanced Programming Methods)/lab2v2/src/Model/ConstantExpression.java
package Model;

public class ConstantExpression implements Expression {

    private int value;

    public ConstantExpression(int value)
    {
        this.value=value;
    }

    public int eval(IDictionary<String,Integer> dict)
    {
        return this.value;
    }

    @Override
    public String toString()
    {
        return ""+this.value;
    }
}

File: MAP(Advanced Programming Methods)/LabAssig7/LabAssig7/Model/Expressions/ConstantExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using LabAssig7.Model.ADT;


namespace LabAssig7.Model.Expressions
{
    public class ConstantExpression:Expression
    {
        private int value;

        public ConstantExpression(int value)
        {
            this.value = value;
        }

        public int Eval(ISymbolTable<string,int> dict)
        {
            return this.value;
        }

        public override string ToString()
        {
            return "" + this.value;
        }
    }
}
